psl-core/src/main/java/org/linqs/psl/reasoner/sgd/SGDReasoner.java

```diff
@@ -123,7 +123,10 @@ public double optimize(TermStore baseTermStore,
         // optimization pass because they are being updated in the variableUpdate() method.
         // Note that the number of variables may change in the first iteration (since grounding may happen then).
         double oldObjective = Double.POSITIVE_INFINITY;
-        float[] oldVariableValues = null;
+        float[] prevVariableValues = null;
+        // Save and use the variable values with the lowest computed objective
+        double lowestObjective = Double.POSITIVE_INFINITY;
+        float[] lowestVariableValues = null;
```

Comment:
It could also be useful debugging info to have the iteration that the lowest objective was reached on.